src/dotnet-interactive-vscode/common/interactiveClient.ts

```diff
@@ -126,7 +127,8 @@ export class InteractiveClient {
             };
 
             let reportOutputs = () => {
-                outputObserver(outputs);
+                outputObserver([...outputs]);
+                outputs = [];
```

Comment:
Not sure we should re-broadcast everything.
Generally with observer pattern, I'd expect to see only the new stuff, not re-emit of everything all over agian (this makes the changes I introduced easier)

Reply:
We support the concept of updating/replacing outputs, so we need to broadcast everything, otherwise 1 display output then 4 updates would be incorrectly rendered as 5 output items.

Reply:
For my benefit can you give me an example of where that would be necessary.
because The following didn't make sense
>  otherwise 1 display output then 4 updates would be incorrectly rendered as 5 output items.

Reply:
@brettfo @jonsequitur 
I personally don't think we should create outputs, instead we should be returning output items.
At least for streams.
I.e if you have `Console.Write(1)`, and `Console.Write(2)` one after the other, 
if we have a single output & in that we have two output items, hen VS Code will render them as `12`.
However if its two sepate outputs each with their own otuput items, then it will be rendered as `1` <New Line> `2`
VSCode has API to append outputs as well.

The current PR tries to work around this by appending them if the previous output has a stream outputitem of the same type as the current one (i.e. concatenate them).

But that's a bigger change, & I didn't want to introduce that here.
I.e. VS Code will render `1` & `2` in the same line if they belong to the same output object.
Currently its all in seprate outputs

Reply:
FYI - this PR does ensure `Console.Write(1)` & `Console.Write(2)` will be rendered as `12`

Reply:
This will fix #799.